.github/workflows/wiki-generate-blueprints.yaml

```diff
@@ -0,0 +1,143 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+name: Total Mayhem Generator for Blueprints
+
+on:
+    workflow_dispatch:
+    workflow_call:
+
+jobs:
+    # Run icon generation for wiki
+    generate-icons:
+      uses: ./.github/workflows/wiki-generate-icons.yaml
+    generate-blueprints:
+        # We need the icon worklow to complete or we wont have an artifact to get the icons from
+        needs: [generate-icons]
+        name: generate
+        permissions: write-all
+        runs-on: ubuntu-latest
+        defaults:
+          run:
+            shell: bash
+
+        steps:
+            # Check out required repositories, sparse where needed
+            - name: Checkout Brewlan Wikigen Repository
+              uses: actions/checkout@v4
+              with:
+                repository: The-Balthazar/BrewWikiGen
+                path: ./brew-wiki-gen
+            
+            - name: Checkout FAF Repository
+              uses: actions/checkout@v4
+              with:
+                repository: FAForever/fa
+                path: fa
+                ref: develop
+                sparse-checkout-cone-mode: |
+                  loc
+                  lua/ui/help/unitscription.lua
+                  lua/ui/help/tooltips.lua
+                  lua/sim/AdjacencyBuffs.lua
+                  lua/system/Blueprints.lua
+
+            - name: Checkout Total Mayhem repository
+              uses: actions/checkout@v4
+              with:
+                  repository: FAForever/fa-total-mayhem
+                  path: mods/fa-total-mayhem
+                  ref: master
+                  sparse-checkout-cone-mode: |
+                      mod_info.lua
+                      wiki
+                      units/*_unit.bp
+                      units/*_LOD0.scm
+                      projectiles
+
+            - name: Checkout FAF Wiki Repository
+              uses: actions/checkout@v4
+              with:
+                repository: FAForever/fa-total-mayhem.wiki
+                path: ./fa-total-mayhem.wiki
+
+            # Flush the current wiki directory. This wont actually wipe anything unless the push at the end of this workflow
+            - name: Flush folder contents
+              run: rm -rf fa-total-mayhem.wiki/*
```

Comment:
This would not preserve any files that are manually written. Is there a reason you flush everything?

Reply:
It was mainly because while doing testing I was hitting issues where files were being left behind then never cleaned up. Since the push wouldn't replace existing files that didn't change I didn't see any negatives and multiple positives in maintaining a stable repeatable output.

But if the idea is to support people manually editing the wiki then I'll remove the flush as this would certainly cause issues in that scenario.